File: src/AnalyticsService/Service/Queries/GetGenericStatsQuery.fs
namespace AnalyticsService.Service.Commands

open AnalyticsService.Service.Model
open MediatR
open System

/// A query for obtaining generic stats about all document batches.
type internal GetGenericStatsQuery(workflowId: Guid) =
    interface IRequest<GeneralBatchStats>
    member this.WorkflowId with get() = workflowId

File: src/AnalyticsService/Service/Queries/GetGenericStatsQuery.cs
using AnalyticsService.Service.Model;
using MediatR;

namespace AnalyticsService.Service.Queries;

/// <summary>
/// A query for obtaining generic stats about all document batches.
/// </summary>
internal sealed record GetGenericStatsQuery(
    Guid WorkflowId
) : IRequest<GeneralBatchStat>;